### PR TITLE
Rails-Blog: Custom methods

### PR DESCRIPTION
**Pull request by ZikaZaki**
# Rails-Blog: Custom methods
## In this pull-request I did the followings:
- [x] Inserted more than 4 users using the `seeds.rb` file.
- [x] Inserted more than 6 posts using the `seeds.rb` file.
- [x] Inserted more than 8 comments using the `seeds.rb` file.
- [x] Inserted more than 6 likes using the `seeds.rb` file.
- [x] Added custom methods as follows:
    -  Users
       - A method that returns the 3 most recent posts for a given user.
    -  Posts
       - A method that updates the posts counter for a user.
       - A method which returns the 5 most recent comments for a given post.
    -  Comments
       - A method that updates the comments counter for a post.
    -  Likes
       - A method that updates the likes counter for a post.
- [x] Made sure that I used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- [x] Made sure that I documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class User < ApplicationRecord
+  def retrieve_recent_posts
+    posts.order(:created_at).limit(3)
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline app/models/user.rb]: - [ ] kindly add all the relations to your user model i.e `has_many :posts, foreign_key: 'author_id'`

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Post < ApplicationRecord
+  belongs_to :user
```
Comment (VICTOR-CHUKWU) [inline app/models/post.rb]: - [ ] a post has many comments and has many likes, kindly include those relations

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @ZikaZaki  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @ZikaZaki, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has failing linters

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Bondok6 — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

## Hi @ZikaZaki 🙋‍♂️, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

TO HIGHLIGHT:

- Great job implementing the changes provided by the previous reviewers ✅
- Well structured files ✅
- Professional README.md file 💯


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 




## Optional suggestions

- [OPTIONAL] The MIT link in your README file is broken, so I suggest to fix it by changing the path to `[MIT](./MIT.md)` instead of `[MIT](./LICENSE)`, and remember to customize your `MIT.md` file 👍

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**